evennia/scripts/taskhandler.py

```diff
@@ -194,10 +446,198 @@ def create_delays(self):
 
         """
         now = datetime.now()
-        for task_id, (date, callbac, args, kwargs) in self.tasks.items():
+        for task_id, (date, callbac, args, kwargs, _, _) in self.tasks.items():
+            self.tasks[task_id] = date, callbac, args, kwargs, True, None
             seconds = max(0, (date - now).total_seconds())
-            deferLater(reactor, seconds, self.do_task, task_id)
+            d = deferLater(self.clock, seconds, self.do_task, task_id)
+            d.addErrback(handle_error)
+            # some tasks may complete before the deferal can be added
+            if self.tasks.get(task_id, False):
+                self.tasks[task_id] = date, callbac, args, kwargs, True, d
 
 
 # Create the soft singleton
 TASK_HANDLER = TaskHandler()
+
+
+class Task:
```

Comment:
I worry that the name `Task` is too generic a name for this class. It's a special helper class for use in the `TaskHandler`, but with such a universal name one may think it's more universal. It would make more sense to rename this `TaskHandlerTask` I think.

Reply:
Done with https://github.com/evennia/evennia/pull/2374/commits/cf8052b816c0011330ead39c85e871c85918a0e4